File: mobile/app/src/main/java/com/remotedesktop/input/PairingCode.kt
package com.remotedesktop.input

object PairingCode {

    private const val ALPHABET = "ABCDEFGHJKLMNPQRSTUVWXYZ23456789"
    private const val LENGTH = 6
    private val regex = Regex("^[" + Regex.escape(ALPHABET) + "]{$LENGTH}$")

    fun normalise(raw: String): String =
        raw.trim().uppercase().filter { it in ALPHABET }.take(LENGTH)

    fun isValid(code: String): Boolean = regex.matches(code)
}

File: mobile/ios/RemoteDesktop/Sources/PairingCode.swift
import Foundation

enum PairingCode {
    static let alphabet = "ABCDEFGHJKLMNPQRSTUVWXYZ23456789"
    static let length = 6

    static func normalise(_ raw: String) -> String {
        let allowed = Set(alphabet)
        let upper = raw.uppercased()
        return String(upper.filter { allowed.contains($0) }.prefix(length))
    }

    static func isValid(_ code: String) -> Bool {
        guard code.count == length else { return false }
        let allowed = Set(alphabet)
        return code.allSatisfy { allowed.contains($0) }
    }
}
